sdk/packet/src/lib.rs

```diff
@@ -224,6 +292,61 @@ impl PartialEq for Packet {
     }
 }
 
+/// A custom implementation of io::Write to facilitate safe (non-UB)
+/// initialization of a MaybeUninit<Packet>
+struct PacketWriter {
```

Comment:
Why not simple wrapper around MaybeUninit packet?

We know the capacity of the buffer, and can determine remaining bytes from the current length and the fixed capacity